src/cache/assignment-cache-factory.ts

```diff
@@ -8,14 +8,21 @@ import { LocalStorageAssignmentCache } from './local-storage-assignment-cache';
 import SimpleAssignmentCache from './simple-assignment-cache';
 
 export function assignmentCacheFactory({
+  forceMemoryOnly = false,
   chromeStorage,
   storageKeySuffix,
 }: {
+  forceMemoryOnly?: boolean;
   storageKeySuffix: string;
   chromeStorage?: chrome.storage.StorageArea;
 }): AssignmentCache {
-  const hasLocalStorage = hasWindowLocalStorage();
   const simpleCache = new SimpleAssignmentCache();
+  if (forceMemoryOnly) {
```

Comment:
instead of doing this, which makes this method more complex (unecessarily, I think), i'd just instantiate `SimpleAssignmentCache` directly from `syncInit`

Reply:
Yea - I don't have strong feelings about it either way. On the one hand this method becomes the central way to instantiate caches without the caller needing to know anything about the `SimpleAssignmentCache` besides saying they want an `inMemory`.

Reply:
i feel like it's an unnecessary abstraction that adds complexity. not gonna block on it, but it's my personal preference ;)